### PR TITLE
Load headers immediately

### PR DESCRIPTION
**Pull request by Valodim**
This PR removes the `throws MessagingException` from `getHeader` family of methods in the Part interface. Those were necessary because LocalMessage loaded headers lazily from the database, which might have thrown. This exception was problematic because there was no good way to deal with it, and the relevant error handling code did everything but handle the error correctly.

The lazy loading "optimization" itself a remnant of the old database structure, where we had a headers table which kept one row per header, which made this query fairly expensive. At this point, we can just join the header field of the root part in when we load all other data of LocalMessage and be done with it.

### CODE REVIEW

File: k9mail-library/src/main/java/com/fsck/k9/mail/store/imap/FetchPartCallback.java [hunk 1]
```diff
@@ -29,8 +28,7 @@ public Object foundLiteral(ImapResponse response,
             String contentType = mPart
                     .getHeader(MimeHeader.HEADER_CONTENT_TYPE)[0];
 
-            return MimeUtility.createBody(literal, contentTransferEncoding,
-                    contentType);
+            return MimeUtility.createBody(literal, contentTransferEncoding, contentType);
```
Comment (cketti) [inline k9mail-library/src/main/java/com/fsck/k9/mail/store/imap/FetchPartCallback.java]: :+1:

File: k9mail-library/src/test/java/com/fsck/k9/mail/internet/MessageExtractorTest.java [hunk 2]
```diff
@@ -46,9 +40,9 @@ public void getTextFromPart_withNoBody_shouldReturnNull() throws Exception {
 
     @Test
     public void getTextFromPart_withTextBody_shouldReturnText() throws Exception {
-        withMimeType("text/plain");
-        withContentType("UTF-8");
-        withBody(createTextBody("Sample text body"));
```
Comment (cketti) [inline k9mail-library/src/test/java/com/fsck/k9/mail/internet/MessageExtractorTest.java]: The general idea was to create a kind of DSL to make tests easier to read.

File: k9mail/src/main/java/com/fsck/k9/mailstore/LocalStore.java [hunk 3]
```diff
@@ -39,6 +39,7 @@
 import com.fsck.k9.mail.MessageRetrievalListener;
 import com.fsck.k9.mail.MessagingException;
 import com.fsck.k9.mail.Store;
+import com.fsck.k9.mail.message.MessageHeaderParser;
```
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/mailstore/LocalStore.java]: This import doesn't seem to be used.

File: k9mail/src/test/java/com/fsck/k9/crypto/MessageDecryptVerifierTest.java [hunk 4]
```diff
@@ -298,8 +300,8 @@ public static Part getPart(Part searchRootPart, int... indexes) {
     }
 
     //TODO: Find a cleaner way to do this
-    private static void addProtocolParameter(Part part) throws MessagingException {
-        String contentType = part.getContentType();
-        part.setHeader(MimeHeader.HEADER_CONTENT_TYPE, contentType + "; protocol=\"application/pgp-encrypted\"");
+    private static void setContentTypeWithProtocol(Part part, String mimeType, String protocol)
```
Comment (cketti) [inline k9mail/src/test/java/com/fsck/k9/crypto/MessageDecryptVerifierTest.java]: Why generalize if you don't make use of the flexibility anywhere?

File: k9mail/src/test/java/com/fsck/k9/message/MessageBuilderTest.java [hunk 5]
```diff
@@ -13,6 +13,7 @@
 import com.fsck.k9.mail.Body;
 import com.fsck.k9.mail.Message.RecipientType;
 import com.fsck.k9.mail.MessagingException;
+import com.fsck.k9.mail.internet.MimeHeader;
```
Comment (cketti) [inline k9mail/src/test/java/com/fsck/k9/message/MessageBuilderTest.java]: Unused import

**Comment by cketti** [comment]
Please try to keep pull requests small and focused on one change. For this one 6969120 + d1cae83 would have sufficed.
I realize it might be a bit more work if you make the changes as part of a larger feature. But reviews will be faster and there's less mental context-switching required when reviewing and addressing review comments.

File: k9mail-library/src/test/java/com/fsck/k9/mail/internet/MessageExtractorTest.java (hunk 2 above)
Comment (Valodim) [inline k9mail-library/src/test/java/com/fsck/k9/mail/internet/MessageExtractorTest.java]: Just using the parts without mocks is so simple these methods ended up as just a bunch of delegated setters, I felt like there was no point in adding an abstraction...

File: k9mail/src/test/java/com/fsck/k9/crypto/MessageDecryptVerifierTest.java (hunk 4 above)
Comment (Valodim) [inline k9mail/src/test/java/com/fsck/k9/crypto/MessageDecryptVerifierTest.java]: To make it obvious inside the tests how the parts are structured. `addProtocolParameter` does not exactly imply "with application/pgp-encrypted protocol to content type header", the calls as they are now show exactly what they do

**Comment by Valodim** [comment]
I kept the unit test changes in here because I thought the other changes depended on them (as in, tests failed otherwise). But that's only true for the following commit which isn't in this PR yet. Will try to be more careful :)

File: k9mail/src/main/java/com/fsck/k9/mailstore/LocalMessage.java [hunk 6]
```diff
@@ -500,47 +490,6 @@ private void deleteMessageRow(SQLiteDatabase db, long messageId) {
         db.delete("threads", "message_id = ?", idArg);
     }
 
-    private void loadHeaders() throws MessagingException {
-        mHeadersLoaded = true;
-        getFolder().populateHeaders(this);
```
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/mailstore/LocalMessage.java]: `LocalFolder.populateHeaders()` is now unused.

File: k9mail/src/test/java/com/fsck/k9/crypto/MessageDecryptVerifierTest.java (hunk 4 above)
Comment (cketti) [inline k9mail/src/test/java/com/fsck/k9/crypto/MessageDecryptVerifierTest.java]: Hm, okay. I would have just changed the method name.

**Comment by cketti** [comment]
:+1: Please rebase and merge.

**Pull request merged**